internal/server/unlink_person_test.go

```diff
@@ -0,0 +1,226 @@
+package server
+
+import (
+	"errors"
+	"github.com/ministryofjustice/opg-sirius-lpa-frontend/internal/sirius"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+)
+
+type mockUnlinkPerson struct {
+	mock.Mock
+}
+
+func (m *mockUnlinkPerson) Person(ctx sirius.Context, id int) (sirius.Person, error) {
+	args := m.Called(ctx, id)
+	return args.Get(0).(sirius.Person), args.Error(1)
+}
+
+func (m *mockUnlinkPerson) UnlinkPerson(ctx sirius.Context, parentId int, childId int) error {
+	return m.Called(ctx, parentId, childId).Error(0)
+}
+
+func TestUnlinkPerson(t *testing.T) {
+	children := []sirius.Person{
+		{
+			ID:         5,
+			Salutation: "Mr",
+			Firstname:  "First",
+			Surname:    "Child",
+			Children:   nil,
+		},
+	}
+
+	person := sirius.Person{
+		ID:           189,
+		UID:          "700000001234",
+		Firstname:    "John",
+		Surname:      "Doe",
+		DateOfBirth:  "1998-09-02",
+		AddressLine1: "123 Somewhere Street",
+		Children:     children,
+	}
+
+	client := &mockUnlinkPerson{}
+	client.On("UnlinkPerson", mock.Anything, 189, 5).Return(nil)
+
+	client.
+		On("Person", mock.Anything, 189).
+		Return(person, nil)
+
+	template := &mockTemplate{}
+	template.
+		On("Func", mock.Anything, unlinkPersonData{
+			Entity: person,
+		}).
+		Return(nil)
+
+	r, _ := http.NewRequest(http.MethodGet, "/?id=189", nil)
+	w := httptest.NewRecorder()
+
+	err := UnlinkPerson(client, template.Func)(w, r)
+	resp := w.Result()
+
+	assert.Nil(t, err)
+	assert.Equal(t, http.StatusOK, resp.StatusCode)
```

Comment:
should probably add `mock.AssertExpectationsForObjects(t, client, template)` to these tests, I know we aren't very good at being consistent with it